File: android/src/main/java/com/MythiCode/camerakit/CameraViewInterface.java
package com.MythiCode.camerakit;

import android.widget.FrameLayout;

import java.util.ArrayList;

import io.flutter.plugin.common.MethodChannel;

public interface CameraViewInterface {

    void initCamera(FrameLayout frameLayout, boolean hasBarcodeReader, char flashMode,
                    boolean isFillScale, ArrayList<Integer> restrictFormat, int cameraSelector);

    void setCameraVisible(boolean isCameraVisible);

    void changeFlashMode(char captureFlashMode);

    void takePicture(String path, final MethodChannel.Result result);

    void pauseCamera();

    void resumeCamera();

    void dispose();

}

File: android/src/main/kotlin/com/MythiCode/camerakit/CameraViewInterface.kt
package com.MythiCode.camerakit

import android.widget.FrameLayout
import io.flutter.plugin.common.MethodChannel
import java.util.*

interface CameraViewInterface {
    fun initCamera(
        frameLayout: FrameLayout?, hasBarcodeReader: Boolean, flashMode: Char,
        isFillScale: Boolean, restrictFormat: ArrayList<Int?>?, cameraSelector: Int
    )

    fun setCameraVisible(isCameraVisible: Boolean)
    fun changeFlashMode(captureFlashMode: Char)
    fun takePicture(path: String?, result: MethodChannel.Result?)
    fun pauseCamera()
    fun resumeCamera()
    fun dispose()
}